plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -606,7 +606,7 @@ def delete_group_rolemapping(self, gid, cid, role_rep, realm="master"):
         """
         available_rolemappings_url = URL_CLIENT_GROUP_ROLEMAPPINGS.format(url=self.baseurl, realm=realm, id=gid, client=cid)
         try:
-            open_url(available_rolemappings_url, method="DELETE", http_agent=self.http_agent, headers=self.restheaders,
+            open_url(available_rolemappings_url, method="DELETE", http_agent=self.http_agent, headers=self.restheaders, data=json.dumps(role_rep),
```

Comment:
Hmm, `DELETE` with `data` is not a really common combination. Do you have a link to some API documentation which documents this?

Reply:
[Keycloak REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html#_client_role_mappings_resource) specifies the `roles` argmument as required. However, without it, there is no error, but all mappings are deleted.